### PR TITLE
fix(controller): Script Output didn't set if template has RetryStrategy

### PR DESCRIPTION
**Pull request by sarabala1979**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [ ] I've signed the CLA.
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).
Fixes: #3981

### CODE REVIEW

File: workflow/controller/operator.go [hunk 1]
```diff
@@ -2174,7 +2174,7 @@ func getStepOrDAGTaskName(nodeName string) string {
 	if strings.Contains(nodeName, ".") {
 		name := nodeName[strings.LastIndex(nodeName, ".")+1:]
 		// Retry, withItems and withParam scenario
-		if indx := strings.LastIndex(name, "("); indx > 0 {
+		if indx := strings.Index(name, "("); indx > 0 {
```
Comment (sarabala1979) [inline workflow/controller/operator.go:2177]: Recent Retry logic refactoring added one more `(` in node name '.divide-by-2(2:3)(0)'. LastIndex won't for that. We should use an Index which will handle retry and non-retry scenario

Comment (alexec) [reply workflow/controller/operator.go:2177]: test?

Comment (sarabala1979) [reply workflow/controller/operator.go:2177]: @alexec added test

**Comment by alexec** [comment]
@sarabala1979 can you sync with master? I don't think we should have the proto differences.

**Review by alexec — APPROVED** [review]
Back-port to v2.11?

**Comment by sarabala1979** [comment]
done

**Pull request merged**